### PR TITLE
update to more recent jenkins with master/slave security.

### PR DESCRIPTION
**Pull request by jtnord**
Update the code to implement newly introduced methods.

### CODE REVIEW

**Comment by jtnord** [comment]
@reviewbybees

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Comment by stephenc** [comment]
:bee:

File: src/main/java/hudson/plugins/distfork/DistForkTask.java [hunk 1]
```diff
@@ -138,4 +141,9 @@ public Object getSameNodeConstraint() {
     public Authentication getDefaultAuthentication() {
         return ACL.SYSTEM;
     }
+
+    @Override
+    public Authentication getDefaultAuthentication(Item item) {
+        return ACL.SYSTEM;
+    }
```
Comment (tfennelly) [inline src/main/java/hudson/plugins/distfork/DistForkTask.java:148]: Why was this added?

Comment (jtnord) [inline src/main/java/hudson/plugins/distfork/DistForkTask.java:148]: @tfennelly new method added to an interface in Jenkins.

Comment (tfennelly) [inline src/main/java/hudson/plugins/distfork/DistForkTask.java:148]: Something from `SlaveToMasterCallable`?

**Comment by tfennelly** [comment]
:bee:

File: src/main/java/hudson/plugins/distfork/DistForkTask.java (hunk 1 above)
Comment (jtnord) [inline src/main/java/hudson/plugins/distfork/DistForkTask.java:148]: [Queue.Task#getDefaultAuthentication(hudson.model.Queue.Item)](http://javadoc.jenkins-ci.org/hudson/model/Queue.Task.html#getDefaultAuthentication%28hudson.model.Queue.Item%29)

**Comment by jtnord** [comment]
@reviewbybees done

**Comment by ghost** [comment]
This pull request has completed our internal processes and we now respectfully request the maintainers of this repository to consider our proposal contained within this pull request for merging.
